src/Pos/Util/UserSecret.hs

```diff
@@ -109,6 +158,7 @@ peekUserSecret path = takeReadLock path $ do
 -- 'writeUserSecretRelease'.
 takeUserSecret :: (MonadIO m) => FilePath -> m UserSecret
 takeUserSecret path = liftIO $ do
+    initializeUserSecret path
```

Comment:
This can race under certain circumstances.

+ this code creates the file with mode 600
+ Attacker (assuming there's write access to the directory) swaps it out with one that doesn't have the same permissions.
+ this code finds it from the path and then writes to it, exposing secrets

I think this is marginal, since probably an attacker can't write to this directory. But it's much better to use the same file handle throughout.